it/uk/gov/hmrc/soletraderidentificationfrontend/controllers/CheckYourAnswersControllerISpec.scala

```diff
@@ -108,50 +111,127 @@ class CheckYourAnswersControllerISpec extends ComponentSpecHelper
       }
     }
   }
+
   "POST /check-your-answers-business" should {
-    "redirect to continue url from the supplied journey config" in {
-
-      await(insertJourneyConfig(
-        journeyId = testJourneyId,
-        continueUrl = testContinueUrl,
-        optServiceName = None,
-        deskProServiceId = testDeskProServiceId,
-        signOutUrl = testSignOutUrl,
-        enableSautrCheck = testEnableSautrCheck
-      ))
-      stubRetrieveSoleTraderDetails(testJourneyId)(status = OK, body = testSoleTraderDetailsJson)
-      stubAuth(OK, successfulAuthResponse())
-      stubMatch(testSoleTraderDetails)(OK, successfulMatchJson(testSoleTraderDetails))
-
-
-      lazy val result = post(s"/identify-your-sole-trader-business/$testJourneyId/check-your-answers-business")()
-
-      result must have(
-        httpStatus(SEE_OTHER),
-        redirectUri(s"$testContinueUrl?journeyId=$testJourneyId")
-      )
+    "redirect to continue url from the supplied journey config" when {
+      "the user has only provided a nino" in {
+        await(insertJourneyConfig(
+          journeyId = testJourneyId,
+          continueUrl = testContinueUrl,
+          optServiceName = None,
+          deskProServiceId = testDeskProServiceId,
+          signOutUrl = testSignOutUrl,
+          enableSautrCheck = false
+        ))
+        stubRetrieveSoleTraderDetails(testJourneyId)(status = OK, body = testSoleTraderDetailsNoSautrJson)
+        stubAuth(OK, successfulAuthResponse())
+        stubMatch(testSoleTraderDetailsNoSautr)(OK, successfulMatchJson(testSoleTraderDetailsNoSautr))
+
+
+        lazy val result = post(s"/identify-your-sole-trader-business/$testJourneyId/check-your-answers-business")()
+
+        result must have(
+          httpStatus(SEE_OTHER),
+          redirectUri(s"$testContinueUrl?journeyId=$testJourneyId")
+        )
+      }
+
+      "the user has provided both a nino and an sautr" in {
+        await(insertJourneyConfig(
+          journeyId = testJourneyId,
+          continueUrl = testContinueUrl,
+          optServiceName = None,
+          deskProServiceId = testDeskProServiceId,
+          signOutUrl = testSignOutUrl,
+          enableSautrCheck = true
+        ))
+        stubRetrieveSoleTraderDetails(testJourneyId)(status = OK, body = testSoleTraderDetailsJson)
+        stubAuth(OK, successfulAuthResponse())
+        stubMatch(testSoleTraderDetails)(OK, successfulMatchJson(testSoleTraderDetails))
+
+
+        lazy val result = post(s"/identify-your-sole-trader-business/$testJourneyId/check-your-answers-business")()
+
+        result must have(
+          httpStatus(SEE_OTHER),
+          redirectUri(s"$testContinueUrl?journeyId=$testJourneyId")
+        )
+      }
```

Comment:
Could we have a test where the config is enabled but the user has skipped the SAUTR entry, and this is valid, please?